io/net/ethtool_test.py

```diff
@@ -74,36 +75,32 @@ def interface_state_change(self, interface, state, status):
         cmd = "ip link set dev %s %s" % (interface, state)
         if process.system(cmd, shell=True, ignore_status=True) != 0:
             return False
-        if status != self.interface_link_status(interface):
+        if not wait.wait_for(configure_network.interface_link_up,
+                             timeout=120, args=[self.iface]):
             return False
         return True
 
-    def interface_link_status(self, interface):
-        '''
-        Return the status of the interface link from ethtool.
-        '''
-        cmd = "ethtool %s" % interface
-        for line in process.system_output(cmd, shell=True,
-                                          ignore_status=True).decode("utf-8") \
-                                                             .splitlines():
-            if 'Link detected' in line:
-                return line.split()[-1]
-        return ''
-
     def test_ethtool(self):
         '''
         Test the ethtool args provided
         '''
-        for state, status in zip(["down", "up"], ["no", "yes"]):
```

Comment:
@vaishnavibhat we were actually running the test when interface is down and up, does the new approach take that into consideration ?

Reply:
Addressed in https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1525